docs/src/main/docs/health/01_health.adoc

```diff
@@ -20,71 +20,191 @@
+ The orchestrator may then take action, such as restarting your application if
+ the health check fails.
+
+A typical health check combines the statuses of all the dependencies that
+ impact availability and the ability to perform correctly:
+
+* network latency
+* storage
+* database
+* other services used by your application
 
 === Prerequisites
 
 Declare the following dependency in your project:
 
-[source,xml,subs="verbatim,attributes"]
+[source,xml]
+----
+<dependency>
+    <groupId>io.helidon.health</groupId>
+    <artifactId>helidon-health</artifactId>
+</dependency>
+----
+
+[source,xml]
+.Optional dependency to use built-in health checks
 ----
-    <dependency>
-        <groupId>io.helidon.health</groupId>
-        <artifactId>helidon-health</artifactId>
-    </dependency>
+<dependency>
+    <groupId>io.helidon.health</groupId>
+    <artifactId>helidon-health-checks</artifactId>
+</dependency>
 ----
 
-To enable the built-in health checks, add the following dependency:
+== API overview
+
+[cols="4,6"]
+.Health check API classes
+|=======
+| `org.eclipse.microprofile.health.HealthCheck`
+| Java functional interface representing the logic of a single health check
+
+| `org.eclipse.microprofile.health.HealthCheckResponse`
+| Result of a health check invocation that contains a state and a description.
+
+| `org.eclipse.microprofile.health.HealthCheckResponseBuilder`
+| Builder class to create `HealthCheckResponse` instances
 
-[source,xml,subs="verbatim,attributes"]
+| `io.helidon.health.HealthSupport`
+| WebServer service that exposes `/health` and invokes the registered health
+ checks
+
+| `io.helidon.health.HealthSupport.Builder`
+| Builder class to create `HealthSupport` instances
+|=======
+
+A health check is a Java functional interface that returns a
+ `HealthCheckResponse` object. You can choose to implement a health check
+ inline with a lambda expression or you can reference a method with the double
+ colon operator `::`.
+
+[source,java]
+.Health check with a lambda expression
 ----
-    <dependency>
-        <groupId>io.helidon.health</groupId>
-        <artifactId>helidon-health-checks</artifactId>
-    </dependency>
+HealthCheck hc = () -> HealthCheckResponse
+        .named("exampleHealthCheck")
+        .up()
+        .build();
 ----
 
+[source,java]
+.Health check with method reference
+----
+HealthCheckResponse exampleHealthCheck(){
+    return HealthCheckResponse
+        .named("exampleHealthCheck")
+        .up()
+        .build();
+}
+HealthCheck hc = this::exampleHealthCheck;
+----
+
+`HealthSupport` is a WebServer service that contains a collection of
+ registered `HealthCheck` instances. When queried, it invokes the registered
+ health check and returns a response with a status code representing the overall
+ state of the application.
+
+[cols="1,5",role="flex, sm7"]
+.Health status codes
+|=======
+| `200` | All health checks succeeded
```

Comment:
Does 200 mean that the "check" operation itself was successful, or that the application is healthy?

Reply:
The application is healthy

Reply:
In that case, I suggest changing the description to: "The application is healthy".